core/blockchain.go

```diff
@@ -1668,6 +1668,11 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals, setHead bool)
 		if parent == nil {
 			parent = bc.GetHeader(block.ParentHash(), block.NumberU64()-1)
 		}
+
+		conversionBlock := uint64(17000) // random block number for now
+		if parent.Number.Uint64() == conversionBlock {
+			bc.StartVerkleTransition(parent.Root, emptyVerkleRoot)
+		}
```

Comment:
This is where the blockchain realizes that the transition should start, and signal a `ForkedDB`/`TransitionTrie` adapter should be used.